src/main/java/Dude.java

```diff
@@ -0,0 +1,159 @@
+                    System.out.println("Marked as not done: " + tasks.get(index));
+                } else if (input.startsWith("delete ")) {
+                    int index = Integer.parseInt(input.substring(7));
+                    if (index < 0 || index >= tasks.size()) {
+                        throw new IndexOutOfBoundsException("Task number does not exist.");
+                    }
+                    Task removedTask = tasks.remove(index);
+                    System.out.println("Deleted: " + removedTask);
+
+                } else {
+                    String[] parts = input.split(" ", 2);
+                    String command = parts[0];
+                    String taskInfo = parts.length > 1 ? parts[1] : "";
+
+                    switch (command.toLowerCase()) {
+                        case "todo":
+                            tasks.add(new ToDo(taskInfo));
+                            System.out.println("Added ToDo: " + taskInfo);
+                            break;
+                        case "deadline":
+                            if (!taskInfo.contains(" /by ")) {
+                                throw new IllegalArgumentException(
+                                        "Invalid deadline format. Use '/by' to specify the deadline.");
+                            }
+                            String[] deadlineParts = taskInfo.split(" /by ", 2);
+                            tasks.add(new Deadline(deadlineParts[0], deadlineParts[1]));
+                            System.out.println("Added Deadline: " + deadlineParts[0]);
+                            break;
+                        case "event":
+                            if (!taskInfo.contains(" /from ") || !taskInfo.contains(" /to ")) {
+                                throw new IllegalArgumentException(
+                                        "Invalid event format. Use '/from' and '/to' to specify the event times.");
+                            }
+                            String[] eventParts = taskInfo.split(" /from ", 2);
+                            String[] eventTimes = eventParts[1].split(" /to ", 2);
+                            tasks.add(new Event(eventParts[0], eventTimes[0], eventTimes[1]));
+                            System.out.println("Added Event: " + eventParts[0]);
+                            break;
+                        default:
+                            System.out.println("Unknown command");
+                            break;
+                    }
+                }
+            } catch (NumberFormatException e) {
+                System.out.println("Error: Please enter a valid task number.");
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println(e.getMessage());
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            } catch (Exception e) {
+                System.out.println("An unexpected error occurred: " + e.getMessage());
```

Comment:
Maybe can combine these three statement into one as they are having the same behaviour?